include/envoy/network/io_handle.h

```diff
@@ -260,16 +260,41 @@ class IoHandle {
   virtual Address::InstanceConstSharedPtr peerAddress() PURE;
 
   /**
-   * Creates a file event that will signal when the io handle is readable, writable or closed.
+   * Duplicates the handle. (see man dup)
```

Comment:
Would it be possible to add a comment about intended use?   dup should only ever be used on listener sockets.

In the future we could try to use a different interface for listen sockets in order to avoid exposing this method for iohandle types that don't support dup.

Reply:
Added a comment in 13ff61610c48391cf70ae52dbc51b2d88c2132dd